evalml/tests/automl_tests/test_automl_iterative_algorithm.py

```diff
@@ -0,0 +1,1146 @@
+        if "Mode Baseline Binary" in row["pipeline_name"]:
+            continue
+        if row["pipeline_name"] == "Name_linear":
+            assert row["parameters"]["Imputer"]["numeric_impute_strategy"] == "mean"
+            assert row["parameters"]["Imputer_1"]["numeric_impute_strategy"] in {
+                "most_frequent",
+                "mean",
+            }
+            assert row["parameters"]["Random Forest Classifier"]["n_estimators"] in {
+                100,
+                125,
+            }
+        if row["pipeline_name"] == "Name_dict":
+            assert row["parameters"]["Imputer"]["numeric_impute_strategy"] in {
+                "most_frequent",
+                "mean",
+            }
+            assert row["parameters"]["Imputer_1"]["numeric_impute_strategy"] in {
+                "median",
+                "mean",
+            }
+            assert row["parameters"]["Random Forest Classifier"]["n_estimators"] in {
+                50,
+                100,
+            }
+
+
+@pytest.mark.parametrize(
+    "allowed_component_graphs",
+    [None, {"graph": ["Imputer", "Logistic Regression Classifier"]}],
+)
+@pytest.mark.parametrize(
+    "pipeline_parameters,set_values",
+    [
+        ({"Logistic Regression Classifier": {"penalty": "l1"}}, {}),
+        (
+            {
+                "Imputer": {"numeric_impute_strategy": "mean"},
+                "Logistic Regression Classifier": {"penalty": "l1"},
+            },
+            {},
+        ),
+        (
+            {
+                "Undersampler": {"sampling_ratio": 0.05},
+                "Logistic Regression Classifier": {"penalty": "l1"},
+            },
+            {"Undersampler"},
+        ),
+        (
+            {
+                "Undersampler": {"sampling_ratio": 0.05},
+                "Oversampler": {"sampling_ratio": 0.10},
+            },
+            {"Undersampler", "Oversampler"},
+        ),
+    ],
+)
+def test_pipeline_parameter_warnings_component_graphs_iterative(
+    pipeline_parameters, set_values, allowed_component_graphs, AutoMLTestEnv, X_y_binary
+):
+    X, y = X_y_binary
+    with warnings.catch_warnings(record=True) as w:
+        warnings.filterwarnings(
+            "always",
+            category=ParameterNotUsedWarning,
+        )
+        automl = AutoMLSearch(
+            X_train=X,
+            y_train=y,
+            problem_type="binary",
+            max_batches=2,
+            n_jobs=1,
+            allowed_component_graphs=allowed_component_graphs,
+            pipeline_parameters=pipeline_parameters,
+            _automl_algorithm="iterative",
+        )
+        env = AutoMLTestEnv("binary")
+        with env.test_context(score_return_value={automl.objective.name: 1.0}):
+            automl.search()
+    assert len(w) == (1 if len(set_values) else 0)
+    if len(w):
+        assert w[0].message.components == set_values
+
+
+def test_graph_automl_iterative(X_y_multi):
```

Comment:
Again this is a test that I don't think we really need to keep (the coverage for this should be in the component graph tests) but if we have it we might as well keep it?